tools/seccomp_util/src/main.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2019 - 2020 ESRLabs
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+use anyhow::Result;
+use itertools::Itertools;
+use regex::Regex;
+use std::{fmt::Write, fs::File, io, io::BufRead, path::PathBuf};
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+struct Opt {
+    /// Path to strace log file
+    input: PathBuf,
+}
+
+fn main() -> Result<()> {
+    env_logger::init();
+
+    let opt = Opt::from_args();
+    generate(opt.input)
+}
+
+fn generate(strace_file: PathBuf) -> Result<()> {
+    let lines = open_file(strace_file);
+    let names = parse_strace(lines);
+    let entry = gen_seccomp_entry(names).expect("Failed to generate seccomp entry");
+    println!("{}", &entry);
+    Ok(())
+}
+
+fn open_file(strace_file: PathBuf) -> Vec<String> {
+    let file = File::open(strace_file).expect("Failed to open strace input file");
+    io::BufReader::new(file)
+        .lines()
+        .filter_map(|s| s.ok())
+        .collect::<Vec<String>>()
+}
+
+fn parse_strace<I>(lines: I) -> Vec<String>
+where
+    I: IntoIterator<Item = String>,
+{
+    // Collect syscall names and arguments
+    let mut names = vec![];
+    // unwrap(): Creating regex from constant expression will never fail
+    let regex = Regex::new(r"^\s*(?:\[[^]]*]|\d+)?\s*([a-zA-Z0-9_]+)\(([^)<]*)").unwrap();
+    for line in lines {
+        if let Some(caps) = regex.captures(line.as_str()) {
+            let name = caps.get(1).map_or("", |m| m.as_str());
+            let _args = caps.get(2).map_or("", |m| m.as_str());
+            names.push(name.to_string());
+        }
+    }
+    // Filter duplicates while retaining order
+    names.into_iter().unique().collect()
+}
+
+fn gen_seccomp_entry<I>(names: I) -> Result<String>
```

Comment:
This could depend on `northstar` with feature `manifest` and use the `serde_yaml` instead of hand crafted yaml printing.  Probably useful when we do here something with the arguments (later).